scim/scim.bal

```diff
@@ -0,0 +1,576 @@
+    Meta meta?;
+    SCIMEnterpriseUser urn\:ietf\:params\:scim\:schemas\:extension\:enterprise\:2\.0\:User?;
+    
+};
+
+public type PatchOp record {
+    string op;
+    Value value;
+    
+};
+
+public type UserPatch record {
+    string[] schemas;
+    PatchOp[] Operations;
+    
+};
+
+public type UserSearch record {
+    string[] schemas;
+    string[] attributes;
+    string filter;
+    string domain?;
+    int startIndex?;
+    int count?;
+    
+};
+
+public type GroupResponse record {
+    int totalResults?;
+    int startIndex?;
+    int itemsPerPage?;
+    string[] schemas?;
+    GroupResource[] Resources?;
+    
+};
+
+public type GroupResource record {
+    string displayName?;
+    Meta meta;
+    Member[] members?;
+    Role[] roles?;
+    string id?;
+    
+};
+
+public type GroupUpdate record {
+    string displayName; 
+    Member[] members?;
+    Meta meta?;
+    Role roles?;
+
+};
+
+public type GroupPatch record {
+    string[] schemas;
+    PatchOpGroup[] Operations;
+    
+};
+
+public type PatchOpGroup record {
+    string op;
+    ValueGroup value;
+    
+};
+
+public type ValueGroup record {
+    Member[] members?;
+    Meta meta?;
+    Role roles?;
+    
+};
+
+public type GroupSearch record {
+    string[] schemas;
+    string filter;
+    int startIndex?;
+
+};
+
+public type Operation record {
+    string method;
+    string path;
+    string bulkId?;
+    SCIMUser|BulkUserUpdate|BulkUserReplace|SCIMGroup|BulkGroupUpdate|BulkGroupReplace data?;
+    
+};
+
+public type Bulk record {
+    int failOnErrors?;
+    string[] schemas;
+    Operation[] Operations;
+    
+};
+
+public type BulkUserReplace record {
+    string[] schemas;
+    string username;
+    Name name;
+    
+};
+
+public type BulkUserUpdate record {
+    string op;
+    string path;
+    string value;
+    
+};
+
+public type BulkGroupReplace record {
+    string displayName;
+    MemberReplace[] members;
+    
+};
+
+public type MemberReplace record {
+    string value;
+    string display;
+    
+};
+
+public type BulkGroupUpdate record {
+    string op;
+    string path;
+    json[] value;
+    
+};
+
+public type OperationResponse record {
+    string method;
+    string location?;
+    string bulkId;
+    Status status;
+    string detail?;
+    OpResponse|string response?;
+    
+};
+
+public type Status record {
+    int code;
+    
+};
+
+public type OpResponse record {
+    string[] schemas; 
+    
+};
+
+public type BulkResponse record {
+    string[] schemas;
+    OperationResponse[] Operations;  
+    
+};
+
+
+public class Client{
+
+    private final string orgName; 
+    private final string clientId;
+    private final string clientSecret; 
+    private final string[] scope;
+    final oauth2:ClientOAuth2Provider provider;
+    final http:Client clientEndpoint;
+    final map<string> headers;
+
+    public isolated function init(string orgName,string clientId, string clientSecret, string[] scope ) returns error? {
```

Comment:
Check whether there are tools like ESLint for Ballerina codes, (a VS code plugin may be) . 
Format the codes with it. 
May be a coding rule set is already available. Please check on this.